src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String DELIMITER = ",";
+    private static final String LS = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] totals = readFromFile(fromFileName);
+        String report = createReport(totals[0], totals[1]);
+        writeToFile(toFileName, report);
+    }
+
+    private int[] readFromFile(String fromFileName) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (line.isBlank()) {
+                    continue;
+                }
+                String[] parts = line.split(DELIMITER);
+                if (parts.length != 2) {
+                    continue;
```

This code currently skips lines where the CSV split does not produce exactly two parts. The requirements state: **"Lines must be split into exactly two parts by a comma; handle malformed lines by throwing a RuntimeException with context (aligns with error handling requirement)."**

Instead of `continue`, keep a `lineNumber` counter while reading and throw a RuntimeException that includes the file name and the failing line number (and preferably the line content). Failing fast here prevents silent data loss and makes debugging easier.